sktime/benchmarking/experiments.py

```diff
@@ -0,0 +1,298 @@
+        train and saves
+    """
+    # Set up the file path in standard format
+    if not overwrite:
+        full_path = (
+            str(results_path)
+            + "/"
+            + str(cls_name)
+            + "/Predictions/"
+            + str(dataset)
+            + "/testResample"
+            + str(resampleID)
+            + ".csv"
+        )
+        if os.path.exists(full_path):
+            build_test = False
+        if train_file:
+            full_path = (
+                str(results_path)
+                + "/"
+                + str(cls_name)
+                + "/Predictions/"
+                + str(dataset)
+                + "/trainResample"
+                + str(resampleID)
+                + ".csv"
+            )
+            if os.path.exists(full_path):
+                train_file = False
+        if train_file is False and build_test is False:
+            return
+
+    # currently only works with .ts
+    trainX, trainY = load_ts(problem_path + dataset + "/" + dataset + "_TRAIN" + format)
+    testX, testY = load_ts(problem_path + dataset + "/" + dataset + "_TEST" + format)
+    if resampleID != 0:
+        trainX, trainY, testX, testY = stratified_resample(
+            trainX, trainY, testX, testY, resampleID
+        )
+    le = preprocessing.LabelEncoder()
+    le.fit(trainY)
+    trainY = le.transform(trainY)
+    testY = le.transform(testY)
+    if clusterer is None:
+        clusterer = set_clusterer(cls_name, resampleID)
+
+    run_clustering_experiment(
+        trainX,
+        clusterer,
+        trainY=trainY,
+        testX=testX,
+        testY=testY,
+        cls_name=cls_name,
+        dataset_name=dataset,
+        results_path=results_path,
+    )
+
+
+def set_clusterer(cls, resampleId=None):
+    """Construct a clusterer.
+
+    Basic way of creating the clusterer to build using the default settings. This
+    set up is to help with batch jobs for multiple problems to facilitate easy
+    reproducability through run_clustering_experiment. You can set up bespoke
+    clusterers and pass them to run_clustering_experiment if you prefer. It also
+    serves to illustrate the base clusterer parameters
+
+    Parameters
+    ----------
+    cls : str
+        indicating which clusterer you want
+    resampleId : int or None, default = None
+        clusterer random seed
+
+    Return
+    ------
+    A clusterer.
+    """
+    name = cls.lower()
+    # Distance based
+    if name == "kmeans" or name == "k-means":
+        return TimeSeriesKMeans(
+            n_clusters=5,
+            max_iter=50,
+            averaging_algorithm="mean",
+            random_state=resampleId,
+        )
+    if name == "kmedoids" or name == "k-medoids":
+        return TimeSeriesKMedoids(
+            n_clusters=5,
+            max_iter=50,
+            averaging_algorithm="mean",
+            random_state=resampleId,
+        )
+
+    else:
+        raise Exception("UNKNOWN CLUSTERER")
+
+
+if __name__ == "__main__":
```

Comment:
could we remove this?

Reply:
sure, I'll put in an example usage instead in the docs